Sources/Widgets/Widgets3D/ImageCroppingWidget/example/index.js

```diff
@@ -125,25 +127,53 @@ actor.getProperty().setDiffuse(0.7);
 actor.getProperty().setSpecular(0.3);
 actor.getProperty().setSpecularPower(8.0);
 
-const cropFilter = vtkImageCropFilter.newInstance();
-cropFilter.setInputConnection(reader.getOutputPort());
-mapper.setInputConnection(cropFilter.getOutputPort());
+mapper.setInputConnection(reader.getOutputPort());
 
 // -----------------------------------------------------------
 // Get data
 // -----------------------------------------------------------
 
+function getCroppingPlanes(imageData, ijkPlanes) {
+  const rotation = quat.create();
+  mat4.getRotation(rotation, imageData.getIndexToWorld());
+
+  const rotateVec = (vec) => {
+    const out = [0, 0, 0];
+    vec3.transformQuat(out, vec, rotation);
+    return out;
+  };
+
+  const [iMin, iMax, jMin, jMax, kMin, kMax] = ijkPlanes;
+  const origin = imageData.indexToWorld([iMin, jMin, kMin]);
+  // opposite corner from origin
+  const corner = imageData.indexToWorld([iMax, jMax, kMax]);
+  return [
+    // X min/max
+    vtkPlane.newInstance({ normal: rotateVec([1, 0, 0]), origin }),
+    vtkPlane.newInstance({ normal: rotateVec([-1, 0, 0]), origin: corner }),
+    // Y min/max
+    vtkPlane.newInstance({ normal: rotateVec([0, 1, 0]), origin }),
+    vtkPlane.newInstance({ normal: rotateVec([0, -1, 0]), origin: corner }),
+    // X min/max
+    vtkPlane.newInstance({ normal: rotateVec([0, 0, 1]), origin }),
+    vtkPlane.newInstance({ normal: rotateVec([0, 0, -1]), origin: corner }),
+  ];
+}
+
 reader.setUrl(`${__BASE_PATH__}/data/volume/LIDC2.vti`).then(() => {
   reader.loadData().then(() => {
     const image = reader.getOutputData();
-    cropFilter.setCroppingPlanes(...image.getExtent());
 
     // update crop widget
     widget.copyImageDataDescription(image);
-    window.asdf = widget;
     const cropState = widget.getWidgetState().getCroppingPlanes();
     cropState.onModified(() => {
-      cropFilter.setCroppingPlanes(cropState.getPlanes());
+      const planes = getCroppingPlanes(image, cropState.getPlanes());
+      mapper.removeAllClippingPlanes();
+      planes.forEach((plane) => {
+        mapper.addClippingPlane(plane);
+      });
+      mapper.modified();
```

Comment:
`addClippingPlane` does not call `publicAPI.modified()`. @jourdain would it be worth it to update `addClippingPlane` to call modified? The only perf case I can think of: not calling `modified` on every add means we don't trigger a render when adding multiple planes.

Reply:
We could have a debounced modified call too?

Reply:
maybe with an optional argument (batch=true)

Reply:
For now I will not bother with making changes to `addClippingPlane`. However, I will make an issue to reference this potential change.